kyaml/runfn/runfn.go

```diff
@@ -169,8 +175,29 @@ func (r RunFns) runFunctions(
 		// the output is nil (reading from Input)
 		outputs = append(outputs, kio.ByteWriter{Writer: r.Output})
 	}
-	err := kio.Pipeline{
-		Inputs: []kio.Reader{input}, Filters: fltrs, Outputs: outputs}.Execute()
+
+	var err error
+	pipeline := kio.Pipeline{
+		Inputs: []kio.Reader{input},
+		Filters: fltrs,
+		Outputs: outputs,
+	}
+	if r.LogSteps {
+		err = pipeline.ExecuteWithCallback(func (op kio.Filter) {
+			var identifier string
+			if execFilter, ok := op.(*exec.Filter); ok {
+				identifier = execFilter.Path
+			} else if containerFilter, ok := op.(*container.Filter); ok {
+				identifier = containerFilter.Image
+			} else {
+				identifier = fmt.Sprintf("%s", op)
```

Comment:
Looks like the `Filter` can be anything. I am not sure if there is any better way to print. Can you take a look? @Liujingfang1

Reply:
The `Filter` is an interface that accepts a list of yaml.RNode and returns a list of filtered yaml.RNode. In this file, the filters are get by `ffp` function if `r.functionFilterProvider` is not set. The `ffp` function covers three types of filters: `container.Filter`, `starlark.Filter` and `exec.Filter`. You also need to over them.

If `r.functionFilterProvider` is set, it may return other types of Filters. Take a look at  https://github.com/kubernetes-sigs/kustomize/tree/master/kyaml/fn/runtime, I think it provides different types of Filters.